docs/federation.md

```diff
@@ -133,22 +138,41 @@ This query comes with the following variables:
       "__typename": "Show",
       "id": 5
     }
-  ]        
-} 
+  ]
+}
 ```
 
-The Reviews DGS needs to implement an `entity fetcher` to handle this query.
-An entity fetcher is responsible for creating an instance of a `Show` based on the representation in the `_entities` query above.
-The DGS framework does most of the heavy lifting, and all we have to do is provide the following:
+The Reviews DGS needs to implement an entity fetcher to handle this query.
+
+- **Entity fetcher**: A method annotated with [`@DgsEntityFetcher`](https://javadoc.io/doc/com.netflix.graphql.dgs/graphql-dgs/latest/com/netflix/graphql/dgs/DgsEntityFetcher.html) that takes a key and returns a single instance of the entity or null.
+
+Our entity fetcher gets the `id` field and returns a `Show` instance:
 
 [Full code](https://github.com/Netflix/dgs-federation-example/blob/master/reviews-dgs/src/main/java/com/example/demo/datafetchers/ReviewsDatafetcher.java)
+
 ```java
 @DgsEntityFetcher(name = "Show")
 public Show movie(Map<String, Object> values) {
-        return new Show((String) values.get("id"), null);
+    String showId = (String) values.get("id")
+    return new Show(showId, null);
+}
+```
+
+In this case, we're not doing any data fetching: our `Show` instance only has an `id` field, and we implement a `Show.reviews` datafetcher in the [next section](#providing-data-with-a-data-fetcher). 
+
+However, we could instead fetch data in the entity fetcher. If our DGS served multiple fields, and they all came from the same data source, we could fetch them all at once in the entity fetcher instead of writing separate datafetchers for each field:
```

Comment:
I would leave this out from the docs - don't think we want to necessarily recommend this as a pattern.

Reply:
So is it an anti-pattern to build the whole entity object? Should we always return the partial entity (only `id` field)?
If it is, maybe it makes sense to mentioned that it is not recommended.

Reply:
IMO it's a valid thing to do, and sometimes the better thing to do, and many won't realize it's an option unless we say it is. Maybe can change/add language around it to say only use it in this circumstance, or "here is why you often don't want to do it?"

Reply:
Ok, I guess if it's providing multiple **federated** fields - so would be good to highlight that.

Reply:
+1 that this is a valid pattern